citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/iterators/CrossRefIterator.java

```diff
@@ -0,0 +1,279 @@
+        typeMap.put("book-series", Publication.BOOK_SERIES);
+        parseCrossRef();
+
+    }
+
+    private void parseCrossRef() {
+
+        articles = new ArrayList<>();
+        for (String doi : info.getDois()) {
+            try {
+                Item item = crossrefService.get(doi);
+                article = new Publication();
+                article.setArticleType(typeMap.get(item.getType())); 
+                article.setJournalMeta(parseJournalMeta(item));
+                article.setArticleMeta(parseArticleMeta(item));
+
+                articles.add(article);
+            } catch (RequestFailedException | IOException e) {
+                logger.error("Couuld not retrieve work for doi: "+ doi, e);
+                // for now we just log the exceptions
+                // we might want to devise a way to decide if the 
+                // service might be down and we should stop sending requests.
+            }
+        }
+    }
+
+    private ContainerMeta parseJournalMeta(Item item) {
+        ContainerMeta meta = new ContainerMeta();
+        meta.setContainerTitle(String.join(", ", item.getContainerTitle()));
+        meta.setPublisherName(item.getPublisher());
+        meta.setPublisherLocation(item.getPublisherLocation());
+        List<Issn> issnList = new ArrayList<Issn>();
+        if(item.getIssnType() != null) {
+            for(IssnType issnType : item.getIssnType()) {
+                Issn issn = new Issn();
+                issn.setIssn(issnType.getValue());
+                issn.setPubType(issnType.getType());
+                issnList.add(issn);
+            }
+        }
+        meta.setIssns(issnList);
+
+        List<Contributor> contributors = new ArrayList<>();
+        if(item.getChair() != null) {
+            Person itemChair = item.getChair();
+            Contributor chair = new Contributor();
+            chair.setContributionType(ContributionType.CHAIR);
+            chair.setGivenName(itemChair.getGiven());
+            chair.setSurname(itemChair.getFamily());
+            chair.setFullName(itemChair.getName());
+            List<Affiliation> affiliations = new ArrayList<>();
+            for(Institution institute: itemChair.getAffiliation()) {
+                Affiliation affiliation = new Affiliation();
+                affiliation.setName(institute.getName());
+                affiliations.add(affiliation);
+            }
+            chair.setAffiliations(affiliations);
+            ContributorId contributorID = new ContributorId();
+            contributorID.setId(itemChair.getOrcid());
+            contributorID.setIdSystem("ORCID");
+            chair.setIds(Arrays.asList(contributorID));
+            contributors.add(chair);
+        }
+        // added Editors & translators to article meta. 
+
+        return meta;
+    }
+
+    private ArticleMeta parseArticleMeta(Item item) {
+        ArticleMeta meta = new ArticleMeta();
+        meta.setArticleTitle(String.join(", ", item.getTitle()));
```

Comment:
are there more than one title?

Reply:
Crossref returns list of titles, so I have joined them here. Don't want to miss any title given to an article.

Reply:
hm, not a great idea. then the title would be different. I would use the first title and then add another field to citesphere "additional-titles" (or something like that) and store additional titles with the extra metadata.

Reply:
?

Reply:
this still